### ISSUE TITLE
[Dialog] Add focus trap to dialog

### ISSUE THREAD

**Issue by akdetrick** [narmi/design_system#359]
Depends on:

- https://github.com/narmi/design_system/issues/185
- https://github.com/narmi/design_system/issues/302

Once `Dialog` has merged and `Button` is a focusable element, add a focus trap to the Dialog for usability. Keyboard navigation should be trapped within the `Dialog` and its contents while it is open.
